lib/modules/ens/index.js

```diff
@@ -265,29 +236,79 @@ class ENS {
           }
           const registryAddress = self.ensConfig.ENSRegistry.deployedAddress;
-          const resolverAddress = self.ensConfig.Resolver.deployedAddress;
           const rootNode = namehash.hash(self.registration.rootDomain);
           const contract = self.ensConfig.FIFSRegistrar;
           contract.args = [registryAddress, rootNode];
-          contract.onDeploy = [
-            `ENSRegistry.methods.setOwner('${rootNode}', web3.eth.defaultAccount).send({from: web3.eth.defaultAccount}).then(() => {
-            ENSRegistry.methods.setResolver('${rootNode}', "${resolverAddress}").send({from: web3.eth.defaultAccount});
-            var reverseNode = web3.utils.soliditySha3(web3.eth.defaultAccount.toLowerCase().substr(2) + '${reverseAddrSuffix}');
-            ENSRegistry.methods.setResolver(reverseNode, "${resolverAddress}").send({from: web3.eth.defaultAccount});
-            Resolver.methods.setAddr('${rootNode}', web3.eth.defaultAccount).send({from: web3.eth.defaultAccount});
-            Resolver.methods.setName(reverseNode, '${self.registration.rootDomain}').send({from: web3.eth.defaultAccount});
-          })`
-          ];
+
           self.events.request('deploy:contract', contract, (err, _receipt) => {
             return next(err);
           });
+        },
+        function registerRoot(next) {
+          let config = {
+            registryAbi: self.ensConfig.ENSRegistry.abiDefinition,
+            registryAddress: self.ensConfig.ENSRegistry.deployedAddress,
+            registrarAbi: self.ensConfig.FIFSRegistrar.abiDefinition,
+            registrarAddress: self.ensConfig.FIFSRegistrar.deployedAddress,
+            resolverAbi: self.ensConfig.Resolver.abiDefinition,
+            resolverAddress: self.ensConfig.Resolver.deployedAddress
+          };
+          async.parallel([
+            function createRegistryContract(paraCb) {
+              self.events.request("blockchain:contract:create",
+                {abi: config.registryAbi, address: config.registryAddress},
+                (registry) => {
+                  paraCb(null, registry);
+                });
+            },
+            function createRegistrarContract(paraCb) {
+              self.events.request("blockchain:contract:create",
+                {abi: config.registrarAbi, address: config.registrarAddress},
+                (registrar) => {
+                  paraCb(null, registrar);
+                });
+            },
+            function createResolverContract(paraCb) {
+              self.events.request("blockchain:contract:create",
+                {abi: config.resolverAbi, address: config.resolverAddress},
+                (resolver) => {
+                  paraCb(null, resolver);
+                });
+            }
+          ], (err, contracts) => {
+            self.ensContract = contracts[0];
+            self.registrarContract = contracts[1];
+            self.resolverContract = contracts[2];
+
+            /*global web3*/
```

Comment:
Feel free to ignore but would it be possible to request web3 from the blockchain connector?

Idealy we would remove this TODO:
https://github.com/embark-framework/embark/blob/develop/lib/core/modules/coderunner/runCode.js#L21-L26

Reply:
I'll try. You're right that using global is bad